lib/api/LogManagerImpl.cpp

```diff
@@ -807,12 +807,53 @@ namespace MAT_NS_BEGIN
     void LogManagerImpl::SetDataInspector(const std::shared_ptr<IDataInspector>& dataInspector)
     {
         LOCKGUARD(m_dataInspectorGuard);
-        m_dataInspector = dataInspector;
+        if(dataInspector == nullptr)
+        {
+            LOG_WARN("Attempting to set nullptr as DataInspector");
+            return;
+        }
+
+        auto itDataInspector = std::find_if(m_dataInspectors.begin(), m_dataInspectors.end(), [&dataInspector](const std::shared_ptr<IDataInspector>& currentInspector)
+        {
+            return strcmp(typeid(*dataInspector).name(),typeid(*currentInspector).name()) == 0;
+        });
+
+        if (itDataInspector != m_dataInspectors.end())
+        {
+            LOG_WARN("Replacing specified IDataInspector with passed in inspector");
+            m_dataInspectors.erase(itDataInspector);
+        }
+
+        m_dataInspectors.push_back(dataInspector);
+    }
+
+    void LogManagerImpl::ClearDataInspectors()
+    {
+        LOCKGUARD(m_dataInspectorGuard);
+        m_dataInspectors.clear();
```

Comment:
Clear doesn't guarantee that the underlying buffer is released. Prefer std::vector()::swap(m_dataInspectors), as this guarantees that the underlying buffer is released.
See: https://www.cplusplus.com/reference/vector/vector/clear/

Reply:
done